payment_payfip/models/payment_provider.py

```diff
@@ -0,0 +1,373 @@
+import logging
+import requests
+import urllib.parse
+from requests.exceptions import ConnectionError
+from xml.etree import ElementTree
+
+from odoo import api, fields, models, _, exceptions
+
+# from odoo.addons.payment.models.payment_acquirer import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class PayFIPAcquirer(models.Model):
+    _inherit = "payment.provider"
+
+    # region Default methods
+    # endregion
+
+    # region Fields declaration
+    code = fields.Selection(selection_add=[("payfip", "PayFIP")], ondelete={"payfip": "set default"})
+
+    payfip_customer_number = fields.Char(
+        string="Customer number",
+        required_if_provider="payfip",
+    )
+    payfip_form_action_url = fields.Char(
+        string="Form action URL",
+        required_if_provider="payfip",
+    )
+    payfip_activation_mode = fields.Boolean(
+        string="Activation mode",
+        default=False,
+    )
+
+    reject_msg = fields.Html(
+        string="Rejected Message",
+        help="The message displayed if the order is rejected during the payment process",
+        default=lambda self: _("Your payment has been rejected."),
+        translate=True,
+    )
+    # endregion
+
+    # region constraint
+    @api.constrains("payfip_customer_number")
+    def _check_payfip_customer_number(self):
+        self.ensure_one()
+        if self.code == "payfip" and self.payfip_customer_number not in ["dummy", ""]:
+            webservice_enabled, message = self._payfip_check_web_service()
+            if not webservice_enabled:
+                raise exceptions.ValidationError(message)
+
+    @api.constrains("state")
+    def _check_environment(self):
```

Comment:
```suggestion
    def _check_payfip_state(self):
```
Rapeller "state" dans le nom de la méthode par convention  + "payfip" pour éviter d'éventuel conflit de nom (model partagé)